src/argilla/client/feedback/schemas/questions.py

```diff
@@ -130,14 +131,15 @@ class _LabelQuestion(QuestionSchema, LabelMappingMixin):
             the list must contain at least two unique labels. Additionally, `labels` can
             also be a dictionary of labels, where the keys are the labels, and the values
             are the labels that will be shown in the UI.
-        visible_labels: The number of visible labels in the UI. Defaults to 20, and must
-            be 3 or greater.
+        visible_labels: The number of visible labels in the UI. Defaults to undefined,
+            which means that it will be automatically set, otherwise it must be either None
+            which means all the labels will be shown, or 3 or greater.
     """
 
     labels: Union[conlist(str, unique_items=True, min_items=2), Dict[str, str]]
-    visible_labels: Optional[conint(ge=3)] = 20
+    visible_labels: Union[Literal["undefined"], conint(ge=3), None] = "undefined"
 
-    @validator("labels", always=True)
+    @validator("labels", pre=True, always=True)
```

Comment:
do we finally need this `pre=True`?

Reply:
Yes, because we ideally want to validate the labels before actually validating everything via the `root_validator`, so that everything that comes into the `root_validator` is previously validated